### ISSUE TITLE
[bug]: Toggle button does not work well on mobile (and confusing on desktop)

### ISSUE THREAD

**Issue by nareshbhatia** [shadcn-ui/ui#4457]
### Describe the bug

The Toggle button has the following background colors for the hover and on states:

- `hover:bg-muted`
- `data-[state=on]:bg-accent`

Unfortunately both these colors map to the same physical color (e.g. `neutral-100` in light mode and `neutral-800` in dark mode). This makes it very difficult to tell if the button is on or off in certain situations. The worst case is the Toggle button on a mobile which is in the off state, but still appears to be on. That's because on the mobile, the button stays in the hover mode even after it is tapped. The issue is less severe on the desktop where the confusion only occurs when you click the button and your cursor is still on it. In this case, even when the button is turned off, it shows the hover color giving the perception that it is on.

The screenshot below is for a outline button on mobile which is turned off, but still looks like it is on:

![IMG_2497](https://github.com/user-attachments/assets/e4d06ebc-0c36-4b01-b6ae-2b083180163d)

You can try this in the docs at https://ui.shadcn.com/docs/components/toggle#outline

### Affected component/components

Toggle, Toggle Group

### How to reproduce

1. Go to https://ui.shadcn.com/docs/components/toggle#outline on a mobile phone
2. Toggle the outline button on
3. Toggle the outline button off

The button appears as if it is on.

### Codesandbox/StackBlitz link

https://ui.shadcn.com/docs/components/toggle#outline

### System Info

```bash
MacOS / iOS
Chrome
```

### Before submitting

- [X] I've made research efforts and searched the documentation
- [X] I've searched for existing issues

**Comment by lucifersosad** [comment]
thank u so much

**Comment by nullnullsieben** [comment]
https://www.instagram.com/1900_officelle?igsh=MW5ueDI0b21qbW44Zg==

lucifersosad ***@***.***> schrieb am Mo., 5. Aug. 2024, 08:08:

**Comment by ammarfaris** [comment]
this is still an issue right?

**Comment by Mirrorgo** [comment]
> this is still an issue right?  

Yes, this is still an issue. I've tried to fix it and submitted a PR, but the official maintainers haven't reviewed it for several months now. However, you can try the solution I've provided by making similar modifications to your local code. You can refer to this link https://github.com/shadcn-ui/ui/pull/4519  for details. Other users have already verified that this approach works.

**Comment by ammarfaris** [comment]
> Yes, this is still an issue. I've tried to fix it and submitted a PR, but the official maintainers haven't reviewed it for several months now. However, you can try the solution I've provided by making similar modifications to your local code. You can refer to this link [#4519](https://github.com/shadcn-ui/ui/pull/4519) for details. Other users have already verified that this approach works.

Works like a charm! Thank you so much @Mirrorgo !

**Comment by shadcn** [comment]
This issue has been automatically marked as stale due to one year of inactivity. It will be closed in 7 days unless there’s further input. If you believe this issue is still relevant, please leave a comment or provide updated details. Thank you. (This is an automated message)

**Comment by shadcn** [comment]
This issue has been automatically closed due to one year of inactivity. If you’re still experiencing a similar problem or have additional details to share, please open a new issue following our current issue template. Your updated report helps us investigate and address concerns more efficiently. Thank you for your understanding! (This is an automated message)
